client/app/bundles/course/material/folders/operations.ts

```diff
@@ -199,9 +203,22 @@ export function updateMaterial(
 
 export function downloadFolder(
   currFolderId: number,
-  onSuccess: () => void,
-  onFailure: () => void,
+  handleSuccess: () => void,
+  handleFailure: () => void,
 ): Operation<void> {
-  return async (_) =>
-    CourseAPI.folders.downloadFolder(currFolderId, onSuccess, onFailure);
+  return async () =>
+    CourseAPI.folders
+      .downloadFolder(currFolderId)
+      .then((response) => {
+        pollJob(
+          response.data.jobUrl,
+          (data: JobCompleted) => {
```

Comment:
Small thing, but I don't think the explicit type `JobCompleted` is needed here. `pollJob` already supplies the function's type, and therefore TypeScript type inference would already know that `data` is a `JobCompleted` _implicitly_.

Reply:
Good catch!